net/netclient/Makefile

```diff
@@ -0,0 +1,56 @@
+# Copyright 2024 Stan Grishin (stangri@melmac.ca)
+# This is free software, licensed under the Apache-2.0 License.
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=netclient
+PKG_VERSION:=0.22.0
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://codeload.github.com/gravitl/netclient/tar.gz/v$(PKG_VERSION)?
+PKG_HASH:=010e7ee72f8cf8ae4cc91b19010b85875d75e70faabd248c992af2487629211d
+
+PKG_MAINTAINER:=Stan Grishin <stangri@melmac.ca>
+PKG_LICENSE:=Apache-2.0
+PKG_LICENSE_FILES:=LICENSE.txt
+
+PKG_BUILD_DEPENDS:=golang/host
+PKG_BUILD_PARALLEL:=1
+PKG_BUILD_FLAGS:=no-mips16
+
+GO_PKG:=github.com/gravitl/netclient
+GO_PKG_BUILD_PKG:=github.com/gravitl/netclient
+GO_PKG_LDFLAGS_X:=main.version=v$(PKG_VERSION)
+
+include $(INCLUDE_DIR)/package.mk
+include ../../lang/golang/golang-package.mk
+
+define Package/netclient
+  SECTION:=net
+  CATEGORY:=Network
+  TITLE:=netclient
+  URL:=https://docs.openwrt.melmac.net/netclient/
```

Comment:
I looked at it and there is no documentation except how I can install the package. This is not sufficient. Cmon.
It even tolds you that you should go to https://github.com/gravitl/netmaker , so your website is kinda useless. Thats it and take it as it is.  I dont think that you will keep it maintained forever. If you want to have something related about how to install the package, thats the OpenWrt documentation for, where everyone can have access.

The URL should be the project URL as it is common it all packages, which we do have here.

Reply:
The project's documentation is obsolete as far as OpenWrt-specific information is concerned, hence the need for OpenWrt-specific information. Unless you're up to collaborating and maintaining the documentation for this project, I suggest you make peace with the fact that the README (and screenshots, etc) are hosted in a different repo (and it is in a publicly available repo should I cease to be able to maintain it) than packages. It hasn't been an issue for many other packages I maintain.

Reply:
I am so sorry, but I disagree with your opinion. Your documentation is not better than anything else. The question is why do you need to have separate OpenWrt specific information? That's why it should be included in upstream documentation (projects or OpenWrt ones wiki) not yours in downstream, when your packages are also in upstream. I doubt that you will have time to maintain your packages in both repositores, keep your documentation up-to-date and there are no useful informations, so I suggest to keep the URL for the project, which is done for all the packages and you should not be exception.

Reply:
> Maintaining the documentation for this project, I suggest you make peace with the fact that the README (and screenshots, etc) are hosted in a different repo

Why I should make peace with that? Everything should be here, not in your downstream repository. Someone can hijack your repository, you might not have 2FA, the third party company with the screenshot might not be available, what about privacy and security? I dont think that you count with that point, so NACK.

Reply:
Should I store screenshots for my documentation in the packages repo?

Also, in regards to "everything should be here", please point me to the acme-acmesh documentation.

> Your documentation is not better than anything else.

It's not better yet, but having it in a different repo allows me to update it without creating PR for packages as well as store screenshots in the repo.

By your logic we shouldn't link anything from the OpenWrt packages repo, because everything else can be hijacked, not have 2FA, etc.

If someone is willing to maintain documentation differently/elsewhere they may create a PR to change the link to something that's better.

> The question is why do you need to have separate OpenWrt specific information? That's why it should be included in upstream documentation (projects or OpenWrt ones wiki)

Same reason there are OpenWrt-specific patches.

Reply:
This should not be marked as resolved conversation, which is not. IMHO.

Your website is not better than I said, so please all URLs and go to use search function points to projects URL, not downstream project.

Reply:
I disagree. I'm willing to continue this conversation with someone who doesn't use f-words when they don't understand something tho.

Reply:
> I'm willing to continue this conversation with someone who doesn't use f-words.

1. All packages points to projects URL (except yours)
2. Documentation says that you need to points it to upstream URL.

And now you are crying that you will do it in on your own. Why they are rules, why there is documentation? Everyone is doing it like it should in standards, but you wants to do it on your own. In my opinion, it is the same like anyone is following laws, but you dont like it. 🤷  So, why are people obey and listening even to best practices, following what should be done. Every single time, there is someone, who wants do it like he wants.

README - where it should be? Here? I am just asking - You want to ship it with the package? Using your words, why "OpenWrt-specifics" couldnt be included in OpenWrt wiki, where even non-Git users can modify/extend/improve it, since OpenWrt specifics are there? In the big picture, take a look (and again the search information and finding it, nothing wrongwith that) how GNU/Linux distributions are doing it. They dont point it lke you.

If is it obsolete like are you saying - then it is time to improve it. Well, how it is different than yours documentation, where you say how to install the package, which are basics? Ok, you got basics users, but in that case, you need to do it more user friendly then pointing them to command-line interface (CLI).